code/iomethods.py

```diff
@@ -267,67 +280,77 @@ def read_data(varName, latName, lonName, userVariables, filelist=None):
+    lonmaxNETCDF = utils.find_nearest(alllonsraw, float(userVariables.LONMAX))
+    latminIndex = (np.where(alllatsraw == latminNETCDF))[0][0]
+    latmaxIndex = (np.where(alllatsraw == latmaxNETCDF))[0][0]
+    lonminIndex = (np.where(alllonsraw == lonminNETCDF))[0][0]
+    lonmaxIndex = (np.where(alllonsraw == lonmaxNETCDF))[0][0]
+
+    # Subsetting the data
+
+    latsraw = alllatsraw[latminIndex:latmaxIndex]
+    lonsraw = alllonsraw[lonminIndex:lonmaxIndex]
+
+    LON, LAT = np.meshgrid(lonsraw, latsraw)
 
     for files in userVariables.filelist:
-        try:
-            thisFile = Dataset(files, 'r', format='NETCDF4')
-            # Clip the dataset according to user lat, lon coordinates
-            # Mask the data and fill with zeros for later
-            tempRaw = thisFile.variables[varName][:, latminIndex:latmaxIndex, lonminIndex:lonmaxIndex].astype('int16')
-            tempMask = ma.masked_array(tempRaw, mask=(tempRaw > userVariables.T_BB_MAX), fill_value=0)
-            # Get the actual values that the mask returned
+        if fileType == 'netCDF':
+            try:
+                thisFile = netCDF4.Dataset(files, 'r', format='NETCDF4')
+                # Clip the dataset according to user lat, lon coordinates
+                # Mask the data and fill with zeros for later
+                tempRaw = thisFile.variables[varName][:, latminIndex:latmaxIndex, lonminIndex:lonmaxIndex].astype('int16')
+                tempMask = ma.masked_array(tempRaw, mask=(tempRaw > userVariables.T_BB_MAX), fill_value=0)
+                # Get the actual values that the mask returned
+
+                # timeIndex, latIndex, lonIndex = index
 
-            # timeIndex, latIndex, lonIndex = index
+                tempMaskedValue = tempMask
+                tempMaskedValue[tempMask.mask] = 0
 
-            tempMaskedValue = tempMask
-            tempMaskedValue[tempMask.mask] = 0
+                xtimes = thisFile.variables[timeName]
 
-            xtimes = thisFile.variables[timeName]
+                # Convert this time to a python datastring
+                time2store, _ = get_model_times(xtimes, timeName)
 
-            # Convert this time to a python datastring
-            time2store, _ = get_model_times(xtimes, timeName)
+                # Extend instead of append because get_model_times returns a list already and we don't
+                # want a list of list
+                timelist.extend(time2store)
+                inputData.extend(tempMaskedValue)
+                thisFile.close()
+            except:
+                print 'bad file! ', files   # TODO Add masking logic to binary files
 
-            # Extend instead of append because get_model_times returns a list already and we don't
-            # want a list of list
-            timelist.extend(time2store)
-            inputData.extend(tempMaskedValue)
-            thisFile.close()
+        elif fileType == 'binary': # TODO Add logic to add time2store to timelist
```

Comment:
isn't there already and option for if fileType == 'binary'? Please explain this if-elif statement